### PR TITLE
Include iterator to source

### PR DESCRIPTION
**Pull request by alarbada**
### Description

Refactors the source connector to use the new snapshot iterator. Adds also a README to document its behaviour. Moves snapshot tests to be from the source instead, so that the snapshot iterator results in an implementation detail.

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-spanner/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Conduit Connector for MySQL
+
+[Conduit](https://conduit.io) connector for google Spanner.
```
Comment (raulb) [inline README.md:3]:
```suggestion
[Conduit](https://conduit.io) connector for Google Spanner.
```

File: common/utils.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	DatabaseName string
+	Endpoint     string
+}
+
+func NewClient(ctx context.Context, config NewClientConfig) (*spanner.Client, error) {
+	options := []option.ClientOption{
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+		option.WithoutAuthentication(),
+	}
+	if config.Endpoint != "" {
```
Comment (raulb) [inline common/utils.go:23]: Are we setting a default endpoint somewhere? If not, I would make this one required and let the validations do the job.

File: destination.go [hunk 3]
```diff
@@ -17,12 +17,7 @@ type Destination struct {
 	config DestinationConfig
 }
 
-type DestinationConfig struct {
-	// Config includes parameters that are the same in the source and destination.
-	Config
-	// DestinationConfigParam must be either yes or no (defaults to yes).
-	DestinationConfigParam string `validate:"inclusion=yes|no" default:"yes"`
-}
+type DestinationConfig struct{}
```
Comment (raulb) [inline destination.go:20]: I'd imagine we're removing this because it was previously unintentionally added?

File: source_config.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+type SourceConfig struct {
+	// Database is the name of the database to use. A valid database name has the
+	// form projects/PROJECT_ID/instances/INSTANCE_ID/databases/DATABASE_ID
+	Database string `json:"database" validate:"required"`
+
+	// Tables represents the spanner tables to read from.
+	Tables []string `json:"tables" validate:"required"`
+
+	// Endpoint is the URL for endpoint override - testing/dry-run only
+	Endpoint string `json:"endpoint"`
```
Comment (raulb) [inline source_config.go:14]: shouldn't this be required?

File: common/utils.go (hunk 2 above)
Comment (alarbada) [reply common/utils.go:23]: It works a bit like the [kinesis connector](https://github.com/conduitio-labs/conduit-connector-kinesis/blob/194ca3afc288a196213917b42e1c54c19dedf688/common/client.go#L49-L58). By default, the google spanner client hits google servers. Having an optional `config.Endpoint` allows me to hit the local docker spanner emulator.

Now, I still haven't end2end tested that my google spanner client usage properly connects to the production servers. I guess it shouldn't, as I'm not using any credentials.

There's an issue already for this #14.

File: destination.go (hunk 3 above)
Comment (alarbada) [reply destination.go:20]: Yup

File: source_config.go (hunk 4 above)
Comment (alarbada) [reply source_config.go:14]: Same as commented before, here's the [kinesis example](https://github.com/conduitio-labs/conduit-connector-kinesis/blob/194ca3afc288a196213917b42e1c54c19dedf688/common/config.go#L24-L25)

**Review by raulb — APPROVED** [review]

**Pull request closed without merging**